packages/lib/src/components/internal/FormFields/Field/Field.tsx

```diff
@@ -1,53 +1,62 @@
-import { FieldProps, FieldState } from './types';
-import { getUniqueId } from '../../../../utils/idGenerator';
 import { ARIA_ERROR_SUFFIX } from '../../../../core/Errors/constants';
+import { useCallback, useRef, useState } from 'preact/hooks';
+import { getUniqueId } from '../../../../utils/idGenerator';
+import { FieldProps } from './types';
 
-class Field extends Component<FieldProps, FieldState> {
-    private readonly uniqueId: string;
-
-    constructor(props) {
-        super(props);
-
-        this.state = { focused: false };
-
-        this.onFocus = this.onFocus.bind(this);
-        this.onBlur = this.onBlur.bind(this);
-
-        this.uniqueId = getUniqueId(`adyen-checkout-${this.props.name}`);
-    }
-
-    onFocus(e) {
-        this.setState({ focused: true }, () => {
-            if (this.props.onFocus) this.props.onFocus(e);
-        });
-    }
-
-    onBlur(e) {
-        this.setState({ focused: false }, () => {
-            if (this.props.onBlur) this.props.onBlur(e);
-            // When we also need to fire a specific function when a field blurs
-            if (this.props.onFieldBlur) this.props.onFieldBlur(e);
-        });
-    }
-
-    static getDerivedStateFromProps(nextProps, prevState) {
-        if (nextProps.focused !== undefined && nextProps.focused !== prevState.focused) {
-            return { focused: nextProps.focused };
-        }
-
-        if (nextProps.filled !== undefined && nextProps.filled !== prevState.filled) {
-            return { filled: nextProps.filled };
-        }
-
-        return null;
-    }
-
-    renderContent({ name, children, errorMessage, helper, inputWrapperModifiers, isLoading, isValid, label, showValidIcon, isCollatingErrors, dir }) {
+const Field: FunctionalComponent<FieldProps> = props => {
+    //
+    const {
+        children,
+        className,
+        classNameModifiers,
+        dir,
+        disabled,
+        errorMessage,
+        helper,
+        inputWrapperModifiers,
+        isCollatingErrors,
+        isLoading,
+        isValid,
+        label,
+        name,
+        onBlur,
+        onFieldBlur,
+        onFocus,
+        onFocusField,
+        showValidIcon,
+        useLabelElement,
+        // Redeclare prop names to avoid internal clashes
+        filled: propsFilled,
+        focused: propsFocused
+    } = props;
+
+    const uniqueId = useRef(getUniqueId(`adyen-checkout-${name}`));
+
+    const [focused, setFocused] = useState(false);
+    const [filled, setFilled] = useState(false);
+
+    // The means by which focussed/filled is set for securedFields
+    if (propsFocused != null) setFocused(!!propsFocused);
+    if (propsFilled != null) setFilled(!!propsFilled);
+
+    // The means by which focussed/filled is set for other fields - this function is passed down to them and triggered
+    const onFocusHandler = useCallback((event: h.JSX.TargetedEvent<HTMLInputElement>) => {
+        setFocused(true);
+        onFocus?.(event);
+    }, []);
```

Comment:
`onFocus` needs to be part of the dependency array here, otherwise `onFocusHandler` won't get updated if the parent component changes the `onFocus` prop reference.

Same for the `onBlurHandler` and `renderContent` dependency arrays

Reply:
We only expect to set `onFocusHandler` and `onBlurHandler` once.
These handlers replace class based code that looked like this:
```
 onFocus(e) {
        this.setState({ focused: true }, () => {
            if (this.props.onFocus) this.props.onFocus(e);
        });
    }
```
`renderContent` has a dependency array

Reply:
Setting up these functions only once does not mean that we shouldn't add their external dependencies to the `useCallback` dependency array. 

This `props.onFocus` prop might get updated at some point from the parent component - maybe this is not the case now, but it can happen in the future. And that will lead the UI to misbehave as we will be using a stale version of the `props.onFocus` inside the `onFocusHandler`. Adding it to the dependency array makes sure that this function will be always up-to-date with its dependencies.

`renderContent` has a dependency array but it is omitting objects like `label`, `onFocusHandler`, etc. which can impact its behavior at some point

Reply:
I see you point but I would argue that the aim of this PR was to do a "one for one" replacement of existing functionality rather than future proofing it

Reply:
The `Field` component defined as class based component does not have issues with stale props because such problem does not exist for this type of component. As it gets migrated to functional component, it has.  That is why I mentioned to use the dependency array.

In my view we would not be future proofing it in this particular case. We would be making it compatible with the current way that it works now as class component.